exist-core/src/main/java/org/exist/xmlrpc/RpcConnection.java

```diff
@@ -732,31 +722,34 @@ public Map<String, Object> getDocumentData(final String docName, final Map<Strin
 
             result.put("data", firstChunk);
             int offset = 0;
-            if (Files.size(tempFile) > MAX_DOWNLOAD_CHUNK_SIZE) {
+            if (tempFile.size() > MAX_DOWNLOAD_CHUNK_SIZE) {
                 offset = firstChunk.length;
 
-                final int handle = factory.resultSets.add(new SerializedResult(tempFile));
+                final int handle = factory.resultSets.add(new CachedContentFile(tempFile));
                 result.put("handle", Integer.toString(handle));
                 result.put("supports-long-offset", Boolean.TRUE);
             } else {
-                temporaryFileManager.returnTemporaryFile(tempFile);
+                tempFile.close();
             }
             result.put("offset", offset);
 
             return result;
         });
     }
 
-    private byte[] getChunk(final Path file, final int offset) throws IOException {
-        final long available = Files.size(file);
+    private byte[] getChunk(final ContentFile file, final int offset) throws IOException {
+        final long available = file.size();
         final int len = (int)Math.min(Math.min(available - offset, MAX_DOWNLOAD_CHUNK_SIZE), Integer.MAX_VALUE);
 
         final byte[] chunk = new byte[len];
-        try (final InputStream is = Files.newInputStream(file)) {
-            is.skip(offset);
+        try (final InputStream is = file.newInputStream()) {
```

Comment:
Should we buffer the I/O here perhaps?

Reply:
@adamretter the buffering is done in the ContentFile itself if not based on memory buffer already...